plugin/pkg/shoot/validator/admission.go

```diff
@@ -1505,27 +1509,38 @@ func validateMachineImagesConstraints(a admission.Attributes, constraints []core
 		for _, machineVersion := range machineImage.Versions {
 			machineImageVersion := fmt.Sprintf("%s:%s", machineImage.Name, machineVersion.Version)
 
-			// update to an expired version is allowed (required for maintenance force updates)
-			if a.GetOperation() == admission.Update || machineVersion.ExpirationDate == nil || machineVersion.ExpirationDate.Time.UTC().After(time.Now().UTC()) {
+			if machineVersion.ExpirationDate == nil || machineVersion.ExpirationDate.Time.UTC().After(time.Now().UTC()) {
 				activeMachineImageVersions.Insert(machineImageVersion)
+			} else if machineVersion.ExpirationDate != nil && machineVersion.ExpirationDate.Time.UTC().Before(time.Now().UTC()) && a.GetOperation() == admission.Update && !isNewWorkerPool {
+				// An already expired machine image version is a viable machine image version for the worker pool iff.
+				//  - this is an update call (no new Shoot creation)
+				//  - updates an existing worker pool (not for a new worker pool)
+				//  - the expired version is higher than the old machine's version
+				// Reason: updating an existing worker pool to an expired machine image version is required for maintenance force updates
+				downgrade, _ := versionutils.CompareVersions(machineVersion.Version, "<", oldMachine.Image.Version)
```

Comment:
Shouldn't we still perform the error check here? We have validation whether the versions can be parsed, but what if `CompareVersions` starts returning errors in other cases?

Reply:
The first argument `machineVersion` comes from the cloud profile and is known to be parseable. The second argument comes from the the machine image of the existing shoot (not the just updated version), which we also know is parseable.
The `CompareVersion` Utils only returns an error in two cases in the subroutine [here](https://github.com/gardener/gardener/blob/187d3e2045e717bd3197ffa924252c4803af37aa/pkg/utils/version/version.go#L78-L77) which can only happen in the `semver.NewConstraint` & `semver.NewVersion`. For the former, we know that the operator `<` and the version are valid, hence there cannot be an error (unless the `semVer` package has a bug) . For the latter, only an error can be returned if the version cannot be parsed which we already established above cannot be the case.

That's why I don't think we need an error check here.

Reply:
After I checked the PR, I saw that we have multiple similar places where we skip the error check :).